doc/api/packages.md

```diff
@@ -495,8 +495,8 @@ specific to least specific as conditions should be defined:
   on native addons. This condition can be disabled via the
   [`--no-addons` flag][].
 * `"node"` - matches for any Node.js environment. Can be a CommonJS or ES
-  module file. _This condition should always come after `"import"` or
-  `"require"`._
+  module file. _In most cases explictily calling out the Node.js platform is
+  not necessary._
```

Comment:
C’mon, gimme just one case where it’s necessary 😄 

You don’t have to add it to the docs, but I’m just very curious now when this could be required.

Reply:
JSPM Core is a good example, see eg https://github.com/jspm/jspm-core/blob/main/package.json#L44. Basically when Node.js really is the only intention (it directs the core modules through `export * from 'node:[mod]'; export { default } from 'node:[mod]'`). This condition comes more into its own when there are more runtimes than just Deno and Node as currently "deno" sufficies as a negative condition anyway in most cases for alternative non-browser platforms.

Reply:
Sounds like maybe we need a condition that's something like "server-side" as in, not browsers. Appropriate for either Node or Deno, maybe other command-line runtimes as well.

Reply:
We have "not browser" and that is the "browser" / "default" condition. All not conditions are well defined. New platform definitions are harder to define in a really concrete way short of being specific platforms.

We did say that "browser" is any web standards API with a DOM, but now that Deno has Node.js compat, perhaps we should consider making "browser" really mean a web browser or web-browser like environment to help ensure this?

Reply:
For that matter, isn't Deno adding support for a DOM? And Node already has JSDom. The line is blurry.

Reply:
Right which is why I'm wondering if maybe we should clarify browser means browser as opposed to trying to define some browser subset. This way not browser is clearer as well. Previously matching "browser" for Deno was a useful compat mechanism but now that we have "deno" and a compat mode for Deno, it seems unnecessary to treat Deno support as matching "browser" conditions (and similarly for other non-browser platforms).

Reply:
I've gone ahead and updated the browser condition clarification here. Let me know how that seems.

Reply:
For DataDog's `dd-trace` package, unfortunately CDNs/bundlers often ignore that it isn't meant to load on the browser and often generate some form of transformation if minor effort is made. For example, we are adding the explicit `node` condition to prevent things like https://cdn.skypack.dev/dd-trace from happening and getting support errors. It also would not work in `deno` or other server envs.

Reply:
Marking this as resolved, but let me know if any wording updates or clarifications can be made.